src/k8s-engine/lima.ts

```diff
@@ -1945,31 +1946,62 @@ CREDFWD_URL='http://${ hostIPAddr }:${ stateInfo.port }'
     await Promise.all(promises);
   }
 
-  async requiresRestartReasons(): Promise<Record<string, [any, any] | []>> {
-    if (this.currentAction !== Action.NONE || this.internalState === K8s.State.ERROR || !this.cfg?.enabled) {
-      // If we're in the middle of starting or stopping, or not using k3s, we don't need to restart.
-      return {};
+  async requiresRestartReasons(settings: Settings['kubernetes']): Promise<Record<string, K8s.RestartReason | undefined>> {
+    const vmConfig = await this.currentConfig;
```

Comment:
I realize it's from an older change, but the definition of `currentConfig` suggests it should have been written as a plain async method called `getCurrentConfig()`